### PR TITLE
lengthAndMask worked out incorrectly

### PR DESCRIPTION
**Pull request by richlanc**
* freerdp do checks against the formal spec, meaning they refused to
work when we calculated the lengthAndMask incorrectly
* lengthAndMask failed to pad each line up to a 2 byte boundary
* Also fixed debug logging in a few places where I needed it

https://osirium.atlassian.net/browse/OS-5192

See https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/71fad4fc-6ad4-4c7f-8103-a442bebaf7d2 for documentation on the padding. 
And https://github.com/FreeRDP/FreeRDP/blob/d96a61d8ca27056845f5cff24efb860bd5211f73/libfreerdp/core/update.c#L423 for where it used to failed in freeRDP.

### CODE REVIEW

**Comment by simon-engledew** [comment]
This fix looks pretty good to me! I wont be able to make time to set up a full environment to test this but as long as you are willing to take 'Last Touch Responsibility' I cant see any reason we don't merge this.

You've checked that this is the same commit the previous known good gold builds are made from?

**Review by simon-engledew — APPROVED** [review]

**Comment by richlanc** [comment]
As long as the xrdp deb whilst built against the tag it suggests it was (1.0.6-osirium0), we're safe. 

```
(ins)0 ✓ richard@richard-osirium ~/projects/freerdp_test_env/xrdp $ git fetch --all
Fetching origin
Fetching upstream
(ins)0 ✓ richard@richard-osirium ~/projects/freerdp_test_env/xrdp $ git log --pretty=oneline upstream/jira/5192_cant_proxy_libfreerdp_through_netrino | head -2
ca832c16f23abbb4bf3642c69d1cae575680fcb1 lengthAndMask worked out incorrectly
595dd1aabd73f7e7ef517bb19bc358ef322ee4c9 Merge pull request #3 from speidy/fetch_timezone_info
(ins)0 ✓ richard@richard-osirium ~/projects/freerdp_test_env/xrdp $ git log --pretty=oneline --tags 1.0.6-osirium0 | head -2
595dd1aabd73f7e7ef517bb19bc358ef322ee4c9 Merge pull request #3 from speidy/fetch_timezone_info
9cef83a133f7f87560a67b0bae8212b811a7e41e Merge branch 'devel' into fetch_timezone_info
```

Yup, happy to take responsibility.

**Comment by simon-engledew** [comment]
It certainly looks like it matches the Dockerfile in Osirium/third-party. 👍 

Nice one chasing this all through and figuring out what it was - brilliant! I can only imagine how much time you must have spent pouring through specs/logs.

I'll let you have the satisfaction of slamming the merge button. Make sure you make an appropriate sound as you hit it. 🎆

**Comment by richlanc** [comment]
I missed the notification for that last message. Thanks for the kind words!

**Pull request merged**
